File: AnySqlWebAdminOld/Code/Math/Trash/cPoint.cs

namespace Vectors
{


    public class Point
    {
        public Point(double x, double y)
        {
            this.x = x;
            this.y = y;
        }

        public Point()
            :this(0,0)
        { }

        public double x;
        public double y;
    }

    public class cPoint
        : Point
    {
        public bool bCurrentlyValid = true;
        public bool bHasInterSection = false;
        public double z = 0;


        //Constructor
        public cPoint(double nXparam = 0, double nYparam = 0, double nZparam = 0)
        {
            this.x = nXparam;
            this.y = nYparam;
            this.z = nZparam;
            this.bCurrentlyValid = true;
        }  // End Constructor


        // instance.ToNormalPoint()
        public Point ToNormalPoint()
        {

            if (this.bCurrentlyValid)
            {
                Point ptReturnValue = new Point();
                ptReturnValue.x = this.x;
                ptReturnValue.y = this.y;
                return ptReturnValue;
            }

            return null;
        }


        // instance.toString();
        public override string ToString()
        {
            string strReturnValue = "(" + this.x + "," + this.y;
            if (z != 0)
                strReturnValue += ", " + this.z + ")";
            else
                strReturnValue += ")";

            if (!this.bCurrentlyValid)
                strReturnValue += " INVALID";

            return strReturnValue;
        }  // End function toString


    } // End cPoint 

} // End Package

File: AnySqlWebAdminOld/TypeScript/Math/cPoint.ts

namespace Vectors
{


    export class Point
    {

        public x: number;
        public y: number;

        constructor(nXparam: number = 0, nYparam: number = 0)
        { }
    }


	export class cPoint extends Point
	{
        public bCurrentlyValid: boolean=true;
		public bHasInterSection:boolean=false;
		public z:number=0;

		
		constructor(nXparam:number=0, nYparam:number=0, nZparam:number=0)
		{
            super(nXparam, nYparam);
			this.x=nXparam;
			this.y=nYparam;
			this.z=nZparam;
			this.bCurrentlyValid=true;
		}  // End Constructor
		
		
		// instance.ToNormalPoint()
		public ToNormalPoint():Point
		{
			let ptReturnValue:Point = new Point();
			if(this.bCurrentlyValid)
			{
				ptReturnValue.x = this.x;
				ptReturnValue.y = this.y;
				return ptReturnValue;
			}
			
			ptReturnValue = null;
			return ptReturnValue;
		}
		
		
		// instance.tostring();
		public tostring():string
		{
			let strReturnValue:string="(" + this.x + "," + this.y;
			if(this.z!=0)
				strReturnValue+= ", " + this.z + ")";
			else
				strReturnValue+= ")";
			
			if(!this.bCurrentlyValid)
				strReturnValue+=" INVALID";
			
			return strReturnValue;
		}  // End function tostring
		
		
	} // End cPoint 
	
} // End Package
